staging/src/k8s.io/apiserver/pkg/cel/library/ip_addrs.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package library
+
+import (
+	"net/netip"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+
+	apiservercel "k8s.io/apiserver/pkg/cel"
+)
+
+// IPAddr provides a CEL function library extension of IP address parsing functions.
+//
+// ipAddr
+//
+// Converts a string to an IP address or results in an error if the string is not a valid IP address.
+// The IP address must be an IPv4 or IPv6 address.
+//
+//	ipAddr(<string>) <IPAddr>
+//
+// Examples:
+//
+//	ipAddr('127.0.0.1') // returns an IPv4 address
+//	ipAddr('::1') // returns an IPv6 address
+//	ipAddr('127.0.0.256') // error
+//	ipAddr(':::1') // error
+//
+// isIPAddr
+//
+// Returns true if a string is a valid IP address. The IP address must be an IPv4 or IPv6 address.
+//
+//	isIPAddr(<string>) <bool>
+//
+// Examples:
+//
+//	isIPAddr('127.0.0.1') // returns true
+//	isIPAddr('::1') // returns true
+//	isIPAddr('127.0.0.256') // returns false
+//	isIPAddr(':::1') // returns false
+//
+// isIPv4 / isIPv6 / isInCIDR
```

Comment:
remove `isInCIDR`?  Looks like this is handled by the below `containsIPAddr`.

Reply:
I started by writing out the docs for what I thought I wanted in this, and changed my mind half way through putting the cidr to ip check on the network address side, so yep, forgot to remove this